handler_models.go

```diff
@@ -235,36 +229,32 @@ func ModelOwnerVersionZip(owner, name string, user *users.User, tx *gorm.DB,
 	if !valid {
 		modelVersion = ""
 	}
+	svc := &models.Service{Storage: globals.Storage}
 
-	model, zipPath, ver, em := (&models.Service{}).DownloadZip(r.Context(), tx,
-		owner, name, modelVersion, user, r.UserAgent())
-	if em != nil {
-		return nil, em
+	zipGetter := res.DownloadZipFile
+	linkRequested := isLinkRequested(r)
+	if linkRequested {
+		zipGetter = res.GetZipLink(svc.Storage)
 	}
 
-	zipFileName := fmt.Sprintf("model-%sv%d.zip", *model.UUID, ver)
-
-	// Remove request header to always serve fresh
-	r.Header.Del("If-Modified-Since")
-	// Set zip response headers
-	w.Header().Set("Content-Type", "application/zip")
-	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", zipFileName))
-	_, err := writeIgnResourceVersionHeader(strconv.Itoa(ver), w, r)
-	if err != nil {
-		return nil, gz.NewErrorMessageWithBase(gz.ErrorUnexpected, err)
+	model, link, ver, em := svc.DownloadZip(r.Context(), tx, owner, name, modelVersion, user, r.UserAgent(), zipGetter)
+	if em != nil {
+		return nil, em
 	}
 
 	// commit the DB transaction
 	// Note: we commit the TX here on purpose, to be able to detect DB errors
-	// before writing "data" to ResponseWriter. Once you write data (not headers)
-	// into it the status code is set to 200 (OK).
+	// before writing "data" to ResponseWriter.
 	if err := tx.Commit().Error; err != nil {
 		return nil, gz.NewErrorMessageWithBase(gz.ErrorZipNotAvailable, err)
 	}
 
-	// Serve the zip file contents
-	// Note: ServeFile should be always last line, after all headers were set.
-	http.ServeFile(w, r, *zipPath)
+	// If ?link=true, fuel will return a link to a cloud storage where the client can perform a subsequent request
+	// to download the resource. If ?link=false or if it is not included, it will serve the file directly to the client.
```

Comment:
Adding the changes included in the `handler_worlds.go` file.

```suggestion
	// If a link was requested, fuel will return a link to a cloud storage where the client can perform a subsequent request
	// to download the resource. If a link was not requested or if it is not included, it will serve the file directly to the client.
```

Reply:
Thanks, I forgot to reflect the change, good catch!